compiler/z/codegen/S390Instruction.cpp

```diff
@@ -2477,12 +2477,20 @@ TR::S390RILInstruction::generateBinaryEncoding()
                }
 #endif
             (*(int32_t *) (cursor + 2)) = boi(i2);
+            TR::Symbol *sym = getSymbolReference()->getSymbol();
+            TR::ResolvedMethodSymbol *resolvedMethodSym = sym->getResolvedMethodSymbol();
+            TR_ResolvedMethod *resolvedMethod = resolvedMethodSym ? resolvedMethodSym->getResolvedMethod() : 0;
             if (getSymbolReference() && getSymbolReference()->getSymbol()->castToMethodSymbol()->isHelper())
```

Comment:
The old code used to check if we had a symbol reference before doing something. The new code however assumes we have a symbol reference. Is this correct?

Reply:
No, we should check if the symref exists. I've done some cleanup in this area: d743c95